### ISSUE TITLE
`https://www.overleaf.com` Experimental Support

### ISSUE THREAD

**Issue by iamhyc** [iamhyc/Overleaf-Workshop#18]
Dear all,

We currently have experimental support for `overleaf.com` from version `v0.2.5`, via Cookies-based login.
Please use `https://www.overleaf.com` as the server address and then [login with cookies](https://github.com/iamhyc/Overleaf-Workshop#how-to-login-with-cookies).

The functionality may be unstable at current point, and your feedback is keenly needed.
Please feel free to post any related problems under this thread.

----
**Milestones**:
- ~~Both `overleaf_session2` and `GCLB` Cookie items are necessary for login, or it will raise "client not handshaken" error;
  The `GCLB` item should be automatically updated during the phase of `socket.io` handshake, but it seems working incorrectly in version `v0.2.5`~~ should be fixed in `v0.2.6`. Thanks to the contribution from @QianrenLi in #20 !
- Since version `v0.3.2`:
  - there will be a default `overleaf.com` server when the server list is empty;
  - password-based login is disabled for `overleaf.com`;
  - a prompt will show when choose cookies-based login;
- Untill version `v0.4.1`, I have heard no report about login for a long time. This functionality (cookie-based login) is considered workable somehow and this issue is closed now.

**Comment by iamhyc** [comment]
@NNCT2904 @risingClouds @LiuWenJia-ops

Hi, just a friendly ping that we currently have experimental support for "overleaf.com".

Please give it a try :)

**Comment by NNCT2904** [comment]
@iamhyc Thank you for notifying me, 
I can login via cookie and see the list of Overleaf projects (pic 1).
However, it seems that I cannot access them at the moment, as they are loading forever (pic 2).
Can you guide me on how to get the log files? I can send them to you to see what happened


<img width="419" alt="Screenshot 2023-09-30 at 1 15 54 pm" src="https://github.com/iamhyc/Overleaf-Workshop/assets/44821915/96063c73-c795-4682-8ef3-381589cffa55">

<img width="419" alt="Screenshot 2023-09-30 at 1 16 26 pm" src="https://github.com/iamhyc/Overleaf-Workshop/assets/44821915/b568cdbd-c0bf-42b5-88b6-3cdc12a5d30f">

**Comment by iamhyc** [comment]
> @iamhyc Thank you for notifying me, I can login via cookie and see the list of Overleaf projects (pic 1). However, it seems that I cannot access them at the moment, as they are loading forever (pic 2). Can you guide me on how to get the log files? I can send them to you to see what happened


Please check the log message in the developer tools ("Help" > "Toggle Developer Tools")

And please also check is your cookie consisted of two items like "overleaf_session2=...; GCLB=..." or just "overleaf_session2=..."?

**Comment by NNCT2904** [comment]
okay I obtained the error message via terminal output host extension

```text 
2023-09-30 13:40:11.811 [error] Error: client not handshaken
	at SocketNamespace.<anonymous> (/Users/ncng/.vscode/extensions/iamhyc.overleaf-workshop-0.2.5/out/api/socketio.js:40:19)
	at EventEmitter.emit [as $emit] (/Users/ncng/.vscode/extensions/iamhyc.overleaf-workshop-0.2.5/node_modules/socket.io-client/lib/events.js:165:15)
	at Socket.publish (/Users/ncng/.vscode/extensions/iamhyc.overleaf-workshop-0.2.5/node_modules/socket.io-client/lib/socket.js:103:19)
	at Socket.onError (/Users/ncng/.vscode/extensions/iamhyc.overleaf-workshop-0.2.5/node_modules/socket.io-client/lib/socket.js:485:10)
	at SocketNamespace.onPacket (/Users/ncng/.vscode/extensions/iamhyc.overleaf-workshop-0.2.5/node_modules/socket.io-client/lib/namespace.js:194:23)
	at Socket.onPacket (/Users/ncng/.vscode/extensions/iamhyc.overleaf-workshop-0.2.5/node_modules/socket.io-client/lib/socket.js:466:30)
	at Transport.onPacket (/Users/ncng/.vscode/extensions/iamhyc.overleaf-workshop-0.2.5/node_modules/socket.io-client/lib/transport.js:98:17)
	at Transport.onData (/Users/ncng/.vscode/extensions/iamhyc.overleaf-workshop-0.2.5/node_modules/socket.io-client/lib/transport.js:69:16)
	at websocket.onmessage (/Users/ncng/.vscode/extensions/iamhyc.overleaf-workshop-0.2.5/node_modules/socket.io-client/lib/transports/websocket.js:75:12)
	at WebSocket.onMessage (/Users/ncng/.vscode/extensions/iamhyc.overleaf-workshop-0.2.5/node_modules/ws/lib/WebSocket.js:442:14)
	at WebSocket.emit (node:events:513:28)
	at Receiver.ontext (/Users/ncng/.vscode/extensions/iamhyc.overleaf-workshop-0.2.5/node_modules/ws/lib/WebSocket.js:841:10)
	at /Users/ncng/.vscode/extensions/iamhyc.overleaf-workshop-0.2.5/node_modules/ws/lib/Receiver.js:536:18
	at Receiver.applyExtensions (/Users/ncng/.vscode/extensions/iamhyc.overleaf-workshop-0.2.5/node_modules/ws/lib/Receiver.js:371:5)
	at /Users/ncng/.vscode/extensions/iamhyc.overleaf-workshop-0.2.5/node_modules/ws/lib/Receiver.js:508:14
	at Receiver.flush (/Users/ncng/.vscode/extensions/iamhyc.overleaf-workshop-0.2.5/node_modules/ws/lib/Receiver.js:347:3)
	at Receiver.finish (/Users/ncng/.vscode/extensions/iamhyc.overleaf-workshop-0.2.5/node_modules/ws/lib/Receiver.js:541:12)
	at Receiver.expectHandler (/Users/ncng/.vscode/extensions/iamhyc.overleaf-workshop-0.2.5/node_modules/ws/lib/Receiver.js:499:31)
	at Receiver.add (/Users/ncng/.vscode/extensions/iamhyc.overleaf-workshop-0.2.5/node_modules/ws/lib/Receiver.js:103:24)
	at TLSSocket.realHandler (/Users/ncng/.vscode/extensions/iamhyc.overleaf-workshop-0.2.5/node_modules/ws/lib/WebSocket.js:825:20)
	at TLSSocket.emit (node:events:513:28)
	at addChunk (node:internal/streams/readable:324:12)
	at readableAddChunk (node:internal/streams/readable:297:9)
	at Readable.push (node:internal/streams/readable:234:10)
	at TLSWrap.onStreamRead (node:internal/stream_base_commons:190:23)
```

The string I inputted to the extension prompt was `overleaf_session2=theCookieFromBrowser`
There is no `GCLB=...` tho

<img width="171" alt="Screenshot 2023-09-30 at 1 42 41 pm" src="https://github.com/iamhyc/Overleaf-Workshop/assets/44821915/eb1c3cee-bee9-4575-b428-785ce8e6e560">

**Comment by iamhyc** [comment]
@NNCT2904 Ok, please firstly visit any of your projects in web browser, and then it should update the `GCLB` one. Then try login again with the updated cookies with two items like `overleaf_session2=...; GCLB=...`.

Also, please make sure your extension version is at least version `0.2.5`.

**Comment by NNCT2904** [comment]
> @NNCT2904 Ok, please firstly visit any of your projects in web browser, and then it should update the `GCLB` one. Then try login again with the updated cookies with two items like `overleaf_session2=...; GCLB=...`.
> 
> Also, please make sure your extension version is at least version `0.2.5`.

Works!!!

**Comment by iamhyc** [comment]
> > @NNCT2904 Ok, please firstly visit any of your projects in web browser, and then it should update the `GCLB` one. Then try login again with the updated cookies with two items like `overleaf_session2=...; GCLB=...`.
> > Also, please make sure your extension version is at least version `0.2.5`.
> 
> Works!!!

Great! This problem has been recorded in `known issues` at the start of this thread.

We will fix this handshake thing later in `v0.3.0` and the Cookies-based login will be more robust then.

**Comment by NikolaSzucsich** [comment]
Hi,

I am using Overleaf Workshop `v0.3.0` (the one avalaible on Visual Studio Marketplace). When I try to login with cookies, login fails with 
![image](https://github.com/iamhyc/Overleaf-Workshop/assets/7926686/0df843d0-d8e6-4bc2-96ea-0190e13c11ef)
I am using overleaf with my university login (`overleaf.com -> log in through your institution`).

Does not work neither with `GCLB` set nor with only `overleaf_session2` set.

Best regards and thanks in advance,
Nikola
